lib/elixir/lib/protocol.ex

```diff
@@ -498,7 +498,7 @@ defmodule Protocol do
 
   defp extract_matching_by_attribute(paths, prefix, callback) do
     for path <- paths,
-        path = to_charlist(path),
+        path = if(is_list(path), do: path, else: String.to_charlist(path)),
```

Comment:
My 2c: a comment here would be worth it, because by reading this I’m afraid the first thought would be "oh with `to_charlist/1` this could be simpler 😄